src/mesh/parallel/shiftedmetric.cxx

```diff
@@ -177,18 +176,24 @@ void ShiftedMetric::shiftZ(const BoutReal *in, const std::vector<dcomplex> &phs,
 }
 
 //Old approach retained so we can still specify a general zShift
-const Field3D ShiftedMetric::shiftZ(const Field3D &f, const Field2D &zangle) {
+const Field3D ShiftedMetric::shiftZ(const Field3D &f, const Field2D &zangle, const REGION region) {
   ASSERT1(&mesh == f.getMesh());
+  ASSERT1(region == RGN_NOX || region == RGN_NOBNDRY); // Never calculate x-guard cells here
```

Comment:
Not sure this is necessary; it may be inefficient to perform shifting in the X guard cells, but shouldn't cause a problem. There are some cases where it may be useful e.g. calculating `D2DXDY`

Reply:
I think the problem I was originally trying to avoid was when shifting in the corner guard cells. I remember getting 'uninitialized value' warnings from Valgrind that I wanted to protect against. They must have been coming from interpolated `zShift` fields though, so probably the right solution is to make sure all points in `zShift`s get initialized to something even when checking is turned off, instead of trying to force the corners to not be used here.

I'll get rid of this check :+1: